### DISCUSSION TITLE
How to reduce hallucination or gibberish in TTS?

### DISCUSSION THREAD

**Discussion by Troyanovsky** [coqui-ai/TTS#2742] [General Q&A]
I'm trying to do TTS with
```
model_name = "tts_models/en/ljspeech/tacotron2-DDC"
# Init TTS
tts = TTS(model_name)

def text_to_speech_file(input_text):
    try:
        tts.tts_to_file(text=input_text, file_path="temp/output.wav")
        return "temp/output.wav"
    except Exception as e:
        print("An error occurred during text to speech conversion:", str(e))
        return None
```
I passed in "Hello, my name is Agent. How can I help you?", but the generated wav file contained gibberish after the sentence "Hello, my name is Agent." with something like "blah blah", and then proceeds to "How can I help you?"

What could be the reason for gibberish or hallucinated output like this? How can I reduce that? Thanks!

**Comment by DrewThomasson** [comment]
No idea on how to reduce it, it seems to be found in all transformer based models, at the moment I'm planning on trying to use https://github.com/linto-ai/whisper-timestamped to get time stamps for each word from the output and compare it to the input text in order to cut out the unwanted hallucinations from the output, cause like, heck

**Reply by ghost** [reply]
That sounds like it could introduce more reliability problems than it solves in an automated scenario, but definately intriguing.
